js/components/Dropdown.jsx

```diff
@@ -34,39 +34,30 @@ var Dropdown = React.createClass({
 		this.initSelect2();
 	},
 
-	componentDidUpdate: function() {
-		this.destroySelect2();
-		this.initSelect2();
-	},
-
 	componentWillUnmount: function() {
 		this.destroySelect2();
```

Comment:
nah, will rename it though. Otherwise the onUpdate event will still hang around even if the unmount the component.